app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save{email.downcase!}
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+   format: {with: VALID_EMAIL_REGEX},
+   uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```

Comment:
Các giá trị như 50 6 255 em khai báo vào file setting nhé
-> tìm hiểu gem config để làm nhé

phân biệt cho chị downcase và downcase! khác nhau thế nào hì

Reply:
Dạ downcase và downcase! đều convert tất cả kí tự trong chuổi về kí tự in thường, tuy nhiên hàm downcase khi thực hiện sẽ coppy ra một biến mới và thực hiện trên biến coppy đó. Còn downcase! sẽ thực thiện trên chính biến đó mà không phải tạo thêm một biến khác.

Reply:
thêm nếu ko có gì để convert thì downcase sẽ trả về nil, còn downcase! sẽ throw exception

(*) note thêm: các method kết thúc với ! sẽ throw exception
các method kết thcus với ? sẽ trả về kết quả true/false nhé

Reply:
Dạ ok chị